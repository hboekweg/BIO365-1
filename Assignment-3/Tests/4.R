source("4.R")

# Declare the test lists
listA = list("A" = 1, "B" = 9, "C" = 4)
listB = list("A" = 4, "B" = -7, "Q" = 3)
listC = list("X" = 4, "Y" = 4, "Z" = 4, "A" = 4)

# Obtain test results
resultA = mergeAndSumLists(listA, listB)
resultB = mergeAndSumLists(listA, listC)
resultC = mergeAndSumLists(listB, listC)

# Check the values in resultA
print(resultA[["A"]])
print(resultA[["B"]])
print(resultA[["C"]])
print(resultA[["Q"]])

# Check the values in resultB
print(resultB[["A"]])
print(resultB[["B"]])
print(resultB[["C"]])
print(resultB[["X"]])

# Check resultC
print(resultC[["A"]])
print(resultC[["B"]])
print(resultC[["Y"]])
print("C" %in% resultC)