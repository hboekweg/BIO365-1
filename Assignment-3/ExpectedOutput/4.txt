[1] 5
[1] 2
[1] 4
[1] 3
[1] 5
[1] 9
[1] 4
[1] 4
[1] 8
[1] -7
[1] 4
[1] FALSE
